File: realworld-back/realworld-common/src/main/java/com/realworld/handler/GlobalExceptionHandler.java
package com.realworld.handler;

import com.realworld.exception.BaseException;
import com.realworld.result.Result;
import io.swagger.v3.oas.annotations.Hidden;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

/**
 * @author YYJYP
 */
@Slf4j
@Hidden
@RestControllerAdvice
public class GlobalExceptionHandler {

	@ExceptionHandler
	public Result exceptionHandler(BaseException ex){
		log.error("异常信息：{}", ex.getMessage());
		return Result.error(ex.getMessage());
	}
}

File: backend-springboot/src/main/kotlin/com/yyjy/config/GlobalExceptionHandler.kt
package com.yyjy.config

import com.yyjy.common.ApiRes
import com.yyjy.common.BusinessException
import io.swagger.v3.oas.annotations.Hidden
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

/**
 * @author Nyxcirea
 * @date 2026/3/2
 * @description: TODO
 */
@Hidden
@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler
    fun handleBusinessException(ex: BusinessException): ApiRes<String?> {
        return ApiRes.fail( ex.message ?: "未知错误")
    }

//    @ExceptionHandler
//    fun handleException(ex: Exception): ApiRes<String?> {
//        return ApiRes.fail( HttpStatus.INTERNAL_SERVER_ERROR.value(), "服务器内部错误: ${ex.message}")
//    }
}